Sources/MongoSwift/MongoCollection+Indexes.swift

```diff
@@ -139,11 +145,12 @@ public struct IndexOptions: Codable {
         self.collation = collation
     }
 
-    // Encode everything besides the name, as we will handle that when encoding the `IndexModel`
+    // The `name` key is encoded in `IndexModel` but needed here for decoding
     private enum CodingKeys: String, CodingKey {
-        case background, expireAfterSeconds, sparse, storageEngine, unique, indexVersion = "v",
+        case background, expireAfterSeconds, name, sparse, storageEngine, unique, version = "v",
```

Comment:
Because `name` is now part of the options encoding/decoding, we should stop doing the weird encoding of `name` in `IndexModel`. Instead, the logic of checking if the user specified the `name` and possibly setting the default should live in the body of `CreateIndexesOperation` (where it probably should have been originally).